qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -141,6 +164,7 @@ def run(self, dag: DAGCircuit) -> DAGCircuit:
         Raises:
             TranspilerError: when the specified synthesis method is not available.
         """
+        dag_bit_indices = {bit: i for i, bit in enumerate(dag.qubits)}
```

Comment:
Paired with the suggestion below at L218 you can skip this step:

```suggestion
```

Reply:
Done in https://github.com/Qiskit/qiskit-terra/commit/11f92363af5b536a0b0ce3444e4e2795b70419e9.